src/test/ui/async-await/pin-needed-to-poll.stderr

```diff
@@ -6,11 +6,9 @@ LL | struct Sleep;
 ...
 LL |         self.sleep.poll(cx)
    |                    ^^^^ method not found in `Sleep`
+  --> $SRC_DIR/core/src/future/future.rs:LL:COL
```

Comment:
Ah! This is a special case I hadn't considered, where the span to note replacement needs to account for being in the main subdiagnostic. This should have either been `:::` instead of `-->` or the note should have appeared as a span_note (`note: ....\n --> ...`).